Prometheus.HttpExporter.AspNetCore/HttpRequestDuration/HttpRequestDurationMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace Prometheus.HttpExporter.AspNetCore.HttpRequestDuration
+{
+    public class HttpRequestDurationMiddleware : HttpRequestMiddlewareBase<Histogram>
+    {
+        public HttpRequestDurationMiddleware(RequestDelegate next, Histogram histogram)
+         : base(histogram)
+        {
+            this.next = next ?? throw new ArgumentNullException(nameof(next));
+            this.requestDuration = histogram;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var stopWatch = new Stopwatch();
+            
+            stopWatch.Start();
+            await this.next(context);
+            stopWatch.Stop();
+
+            var labelData = GetLabelData(context);
+            
+            if (labelData != null) {
```

Comment:
It is unclear to me what the purpose of the if statement is. As far as I can tell, GetLabelData() never returns null.